tests/filesystem_tests/test_hdfs_handler.py

```diff
@@ -53,6 +56,36 @@ def test_klist_not_exist(self):
 
         os.environ['PATH'] = path
 
+    def create_dummy_keytab(self, tmpd, dummy_username):
+        dummy_password = "123"
+        keytab_path = os.path.join(tmpd, "user.keytab")
+        command = "(echo 'addent -password -p {}@{} -k 1 -e rc4-hmac' &&\
+                  sleep 1 && echo {} && echo write_kt {}) \
+                  | ktutil".format(dummy_username, "dummy_realm",
+                                   dummy_password, keytab_path)
+        pipe = subprocess.Popen(command, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE, shell=True)
+        out, err = pipe.communicate()
+        return keytab_path
+
+    def test_get_principel_name_from_keytab(self):
+        dummy_username = "IAmADummy"
+        with tempfile.TemporaryDirectory() as tmpd:
+            # save the original KRB5_KTNAME
+            original_krb5_ktname = os.environ.get('KRB5_KTNAME')
+            try:
+                keytab_path = self.create_dummy_keytab(tmpd, dummy_username)
+
+                os.environ['KRB5_KTNAME'] = keytab_path
+                with HdfsFileSystem() as handler:
```

Comment:
My intention was not to change from `create_handler` to direct object construction, but to clarify what unit feature to be tested with the test method name matching test target. Hereby there're several unit functionalities here, 1) calling the klist command, 2) parsing klist command output, 3) identifying the user name. Strictly speaking, all these three must be tested one by one in unit tests. I don't request strict tests, but at least it must be clarified what is being tested here.

In other word, `_parse_klist_output()` is tested at (new) line 93 below, but why not `_parse_klist_keytab_output(out)` ? Also integrated test of those would be nice if it's done here.